tests/core/test_variable.py

```diff
@@ -0,0 +1,472 @@
+# -*- coding: utf-8 -*-
+
+import datetime
+
+from openfisca_core.model_api import Variable
+from openfisca_core.periods import MONTH, ETERNITY
+from openfisca_core.columns import IntCol
+
+import openfisca_country_template as country_template
+from openfisca_country_template.entities import Person
+
+# Check which date is applied whether it comes from Variable attribute (end)
+# or formula(s) dates.
+
+
+tax_benefit_system = country_template.CountryTaxBenefitSystem()
+
+
+# HELPERS
+
+
+def new_simulation(tax_benefit_system, month):
+    return tax_benefit_system.new_scenario().init_from_attributes(
+        period = month,
+        input_variables = dict(
+            ),
+        ).new_simulation()
+
+
+def vectorize(function, number):
+    return (function.zeros() + 1) * number
+
+
+def check_error_at_add_variable(tax_benefit_system, variable, assertion_message_prefix):
+    try:
+        tax_benefit_system.add_variable(variable)
+    except AssertionError, e:
+        if e.message:
+            assert e.message.startswith(assertion_message_prefix), e.message
+        else:
+            raise
+    except:
+        raise
+
+
+# TESTS
+
+
+# NO DATED ATTRIBUTE(S) - NO DATED FORMULA
+
+
+class variable__no_date(Variable):
+    column = IntCol
+    entity = Person
+    definition_period = MONTH
+    label = u"Variable without date."
+
+
+tax_benefit_system.add_variable(variable__no_date)
+
+
+def test_variable__no_dated_attributes():
+    variable = tax_benefit_system.column_by_name['variable__no_date']
+    assert variable is not None
```

Comment:
This can't be `None`, otherwise `variable = tax_benefit_system.column_by_name['variable__no_date']` would have thrown an error.